classes/Validate.php

```diff
@@ -156,6 +156,22 @@ public static function isImageSize($size)
         return preg_match('/^[0-9]{1,4}$/', $size);
     }
 
+    /**
+     * Check whether given customer name is valid
+     *
+     * @param string $name Name to validate
+     *
+     * @return int 1 if given input is a name, 0 else
+     */
+    public static function isCustomerName($name)
+    {
+        $validityPattern = Tools::cleanNonUnicodeSupport(
+            '/^(?:[^0-9!<>,;?=+()\/\\@#"°*`{}_^$%:¤\[\]|\.。]|[\.。](?:\s|$))*$/u'
```

Comment:
hi,
i take a look at our real customers databases, and it appears that some customers use the dot as separator like "jean.marc" or "J.R." ,
if there is no sql upgrade , we will have some customers blocked during some objects loads or saving.

regards

Reply:
I did mentionned this case and the fact it should be inserted in an upgrade task on the other thread 
https://github.com/PrestaShop/PrestaShop/pull/13559#issuecomment-487051723
`update ps_customer set firstname = replace(replace(firstname,'.','. '),' ',' ');`
`update ps_customer set lastname = replace(replace(lastname,'.','. '),' ',' ');`

Reply:
I dont think a database upgrade is the right thing to do. Imagine this: you update to the latest patch version of Prestashop that states "Protects you against spam" (so you expects this patch to change *nothing*  in your regular shop behavior*) and you find out that this patch modified your database behind your back ?

What about the people that use the lastname as an identifier (because they have nothing better) for a synchronizing system between their shop and their ERP ?

Not to forget the cost of this query on very large databases, might trigger timeouts or deadlocks.

That's why the sql upgrade task does not look a suitable solution for me.

*Which should always be the case for patch versions, but sometimes it cant be avoided. But not here.

Reply:
cost of the query? Just applied those 2 queries on 272455 customer in a breeze, less than 800ms

Reply:
I agree with @matks in that it's risky to change existing data without warning, not to mention that the process could potentially be too long to perform during the autoupgrade.

I propose adding a release note explaining the potential problem and suggesting to run similar queries manually when upgrading -- and I insist on the "similar" because `replace('John B. B., '.', '. ')` returns `John B.  B. `, and `regexp_replace` is only available on MySql 8+